hack/tools/check-go-version.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+GO_MAJOR=${GO_MAJOR:-1}
+GO_MINOR=${GO_MINOR:-22}
```

Comment:
Require the "desired go version" to be passed in so we don't have overrides and defaults in two places?